content/news/XXXX-XX-XX-dvs-internals-pt1.md

```diff
@@ -0,0 +1,156 @@
+
+There are many formats out there:
+[VirtualDJ](http://www.virtualdj.com/buy/controlvinyl.html), [Rekordbox](https://www.pioneerdj.com/product/features/software/rekordbox-dvs-control-vinyl/), [Numark](https://www.numark.com/product/virtualvinyl), [MixVibes](https://www.mixvibes.com/cross-dj-4/), [Traktor](https://www.native-instruments.com/en/products/traktor/digital-vinyl/traktor-scratch-a10/) and [Serato](https://serato.com/dj/pro/expansions/dvs?dvs=dvs-ready) all provide their own solution.
+Mixxx (and the underlying [xwax](https://xwax.org/) library) [support the latter 3 formats](https://manual.mixxx.org/2.3/en/chapters/vinyl_control.html#supported-timecode-media).
+
+Serato is one of the most popular and robust digital vinyl systems.
+There's also Serato Control CD which basically works the same, just for CDJs instead of turntables.
+Since the latter essentially works the same as the vinyl version and can be [downloaded for free from the Serato Website](https://serato.com/controlcd/downloads), I'm going to explain how it works based on that format.
+
+Let's take a look at the timecode signal.
+If you open the WAV file with an audio editor like Audacity, it looks like this:
+
+![Serato CD Timecode Signal in Audacity]({static}/images/news/timecode-signal.png)
+
+As you can see, the timecode signal is a stereo signal.
+The left channel is on top and the right channel is on the bottom.
+Both channels look very similar, but the right channel is shifted a little.
+
+## Detecting the Playback Velocity
+
+Let's focus on the left channel for now.
+The wave basically looks similar to a sine wave that you may remember from your high school math classes.
+You can divide that signal into "cycles", where each cycle starts from zero and goes into a positive part (above the black line) followed by a negative part (below the black line) and then repeats:
+
+![Cycle of left channel in timecode signal]({static}/images/news/timecode-signal-cycle.png)
+
+Each cycle has the same length and if we count the number of cycles in a single second of audio, there are 1000 of them.
+Hence, we now know that the signal has a frequency of 1000 Hz (Hz = 1/s, so it's basically a fancy way saying "per second").
+
+With that information, the DVS system can detect how fast the record is playing by checking how many cycles there are in a second.
+If there are 2000 cycles per second, it's playing a double speed, at 500 cycles per second the record is playing at half speed, and so on.
+
+But how does the computer know how many cycles there are in a second?
+We can't actually wait for a second, because the delay (also called latency) would be very noticeable and make the system unsuitable for scratching.
+
+## Analog vs. Digital Audio
```

Comment:
Shouldn`t this be moved before the "Detecting the Playback Velocity" paragraph? 

You can also mention that the signal you see in Audacity is already digital, and mention that you need to zoom in to make it visible. 

Than you can close the loop when talking later about noise in an real vinyl system.

Reply:
> Shouldn`t this be moved before the "Detecting the Playback Velocity" paragraph?

Unsure. I only added that paragraph because it's necessary to understand how zero crossing detection works. This is not supposed an in-depth guide to DSP. If I move it below the "Basics" section, then readers might wonder why that knowledge is needed and may have already forgotten it at the point where it is.

For the same reason, I don't think it's necessary to explain noise levels here. It surely is necessary for implementation, but if you just want to get the basic understanding, it might just confuse the reader.

Reply:
This was just an idea when reading this article. I am probably not the typical reader ...